packages/core/src/shared/utilities/timeoutUtils.ts

```diff
@@ -387,3 +387,21 @@ export class TimeLag {
         globals.clock.clearInterval(this.intervalRef)
     }
 }
+/**
+ * Tries to execute an asynchronous Promise with a specified time limit.
+ * If the Promise is not resolved within the time limit, it returns `undefined`.
+ *
+ * @param {Promise<T>} asyncPromise - The asynchronous Promise to execute.
+ * @param {number} timeLimit - The time limit in milliseconds.
+ * @returns {Promise<T | undefined>} A Promise that resolves with the result of the original Promise if it completes within the time limit, or `undefined` if the time limit is exceeded.
+ */
+export function tryTimeout<T>(asyncPromise: Promise<T>, timeLimit: number): Promise<T> {
```

Comment:
I think you may be able to just use `waitUntil()` with the `timeout` value set to the time limit

Reply:
This would be preferable

Reply:
Fixed